plugins/lookup/lookup.py

```diff
@@ -294,23 +298,37 @@ def run(self, terms, variables=None, **kwargs) -> list[str]:
             state_file_dir,
         )
 
+    @staticmethod
+    def sanitize_urls(*args):
+        sanitized_inputs = []
+        for input_var in args:
+            input_var = str(input_var).strip()
+            input_var = input_var.rstrip("/")
+            sanitized_inputs.append(input_var)
+        return sanitized_inputs
+
     @staticmethod
     def get_urls(base_url: str, api_url: str, identity_url: str) -> tuple[str, str]:
-        if base_url:
-            base_url = base_url.rstrip("/")
+        if api_url == BITWARDEN_API_URL and identity_url != BITWARDEN_IDENTITY_URL:
+            # unset the default API URL so that the error message is correct.
+            # otherwise, the user will see the default API URL in the error message,
+            # which is confusing
+            api_url = None
+            display.error(API_IDENTITY_URL_ERROR.format(api_url, identity_url))
+            raise AnsibleError(API_IDENTITY_URL_ERROR.format(api_url, identity_url))
+        elif api_url != BITWARDEN_API_URL and identity_url == BITWARDEN_IDENTITY_URL:
+            # unset the default Identity URL so that the error message is correct.
+            # otherwise, the user will see the default Identity URL in the error message,
+            # which is confusing
+            identity_url = None
+            display.error(API_IDENTITY_URL_ERROR.format(api_url, identity_url))
+            raise AnsibleError(API_IDENTITY_URL_ERROR.format(api_url, identity_url))
+        elif base_url != BITWARDEN_BASE_URL:
```

Comment:
What about having the default for `base_url` be nothing? 

It seems weird to have a default value and ignore it.

_____________________________________________________________________

If the user sets a `base_url` and one of the two `api_url` and `identity_url`.

```
base_url = "https://vault.example.com"
api_url = "https://vault.example.com/api"
```

I think this would error on checks above? 

This would contradict the statement on `base_url`.

```
 description: Base url to use. If provided, api_url and identity_url will be **ignored.**
```

Reply:
I left the default base url const, but fixed behavior contradiction in b60562f8d36e7cefb1a44ba4264b50d3a99b2389.